src/app/core/services/filters.service.ts

```diff
@@ -34,13 +35,47 @@ export const DEFAULT_FILTER: Filter = {
  * Filters are subscribed to and emitted from this service
  */
 export class FiltersService {
+  readonly presetViews: {
+    [key: string]: () => Filter;
+  } = {
+    currentlyActive: () => ({
+      title: '',
+      status: ['open pullrequest', 'merged pullrequest', 'open issue', 'closed issue'],
+      type: 'all',
+      sort: { active: 'status', direction: 'asc' },
+      labels: [],
+      milestones: this.getMilestonesForCurrentlyActive().map((milestone) => milestone.title),
+      hiddenLabels: new Set<string>(),
+      deselectedLabels: new Set<string>()
+    }),
+    contributions: () => ({
+      title: '',
+      status: ['open pullrequest', 'merged pullrequest', 'open issue', 'closed issue'],
+      type: 'all',
+      sort: { active: 'id', direction: 'desc' },
+      labels: [],
+      milestones: this.milestoneService.milestones.map((milestone) => milestone.title),
+      hiddenLabels: new Set<string>(),
+      deselectedLabels: new Set<string>()
+    }),
+    custom: () => this.filter$.value
+  };
```

Comment:
Since changing labels, hiddenLabels, deselectedLabels, sort and title do not make the preset view "custom", should we reset them or set them to a fixed value when setting the preset view?
To me, it is unintuitive that since it is "not part of the preset views" since changing these parameters did not affect the status of my preset view being put into custom, choosing a preset view should not reset these parameters.

Reply:
Additionally, is it intentional to provide currentlyActive preset with an invalid sort id of status?

Reply:
I would be more inclined to standardise the preset views. The purpose of "currently active" preset view, for e.g., is to get the distribution of work between people within the current milestone. The user might forget to clear all label filters before setting the preset view to "currently active", and thereby get the wrong impression of the work distribution.

Reply:
I see and that makes sense. Is there a reason the other filters aside from the labels are not "tracked" by the preset views?

Reply:
I would argue that the more important part of the preset view is to get the total count. So for the other filters not being tracked:

* Title: the user might want to search for a particular issue/PR. Technically, the user still stays in the same preset view, as the user is still in the same preset view when the search input is cleared.
* Sort: arguably, sort is an important part of the filters, especially for currently active, but user still can choose their own sorting order, and the total count of each column does not change.
* Labels, selected labels, deselected labels: as explained above, we can look into including these as one of the "tracked" filters.
* Hidden labels: This only affects which label is shown on each PR/issue card and which is not, so total count of each column does not change.

These are just my reasoning, let me know if you think otherwise.

Reply:
Your reasoning is sound. I am inclined to agree but still think there may be some confusion.
When I was testing it, I found it strange that my preset views were maintained when i did something like search by title yet my title was reset when i chose a preset view.
Perhaps a new feature could encompass making a (edited) status for the current preset view when these filters are changed while remaining in the preset view.
However, this would be in a separate PR, I have no issue with the current implementation thank you for clarifying your thought process.

Reply:
I have found out an easy way to let the filters service track the labels fields as well. Because `sanitiseLabels` is only called whenever the labels are polled, I make sure that the update of filters from this method does not change the preset view value.